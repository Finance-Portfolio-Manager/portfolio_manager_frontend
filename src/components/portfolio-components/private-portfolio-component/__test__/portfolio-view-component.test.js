import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import PortfolioView from './../portfolio-view-component';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let stock1 = {stockName:"Amazon", stockQuantity:10, stockAveragePrice:3300, stockCurrentPrice:3400, stockChange:8};
let stock2 = {stockName:"Tesla", stockQuantity:15, stockAveragePrice:700, stockCurrentPrice:715, stockChange:2};
let stock3 = {stockName:"AMC", stockQuantity:20, stockAveragePrice:10, stockCurrentPrice:33, stockChange:230};
let stockList = [stock1, stock2, stock3];

const wrapper = shallow(<PortfolioView tableRows={ stockList }></PortfolioView>);

describe("portfolio view rendering", ()=>{
    test("matches snapshot", ()=>{
        expect(wrapper).toMatchSnapshot();
    })

    test("heading renders", ()=>{
        // console.log(wrapper.debug());
        expect(wrapper.find('h4').length).toBe(1);
    })

    test("table rows render", ()=>{
        expect(wrapper.find("tr").length).toBe(4);
    })
})
