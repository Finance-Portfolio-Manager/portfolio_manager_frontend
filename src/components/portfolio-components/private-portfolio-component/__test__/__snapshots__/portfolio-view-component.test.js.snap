// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`portfolio view rendering matches snapshot 1`] = `ShallowWrapper {}`;
